src/http/server/handlers/static_file_handler.cr

```diff
@@ -99,9 +101,55 @@ class HTTP::StaticFileHandler
         end
       end
 
-      context.response.content_length = file_info.size
       File.open(file_path) do |file|
-        IO.copy(file, context.response)
+        if (range = context.request.headers["Range"]?) && (match = range.match(/bytes=([0-9\-,]+)/))
+          ranges = match[1].split(',').map do |range|
+            start, finish = range.strip.split('-', 2)
```

Comment:
This parsing mechanism looks like it doesn't do proper error handling. It breaks on invalid range values such as `bytes=1` or `bytes=1-2-3`.
Also, there is some duplication between matching the overall (incomplete) format via regex and then splitting into substrings.

Reply:
> This parsing mechanism looks like it doesn't do proper error handling. It breaks on invalid range values such as `bytes=1` or `bytes=1-2-3`.

This is not actionable feedback. Define “breaks”. What do you think the expected and actual behaviors are here?

> Also, there is some duplication between matching the overall (incomplete) format via regex and then splitting into substrings.

What do you mean here?

Reply:
Additionally, how likely is this header to be malformed? How critical is it to get this precisely correct before the feature is merged at all? If it has to be perfect before it gets merged I’ll just close the PR and publish it as a shard.

Reply:
`bytes=1` raises `IndexError` and `bytes=1-2-3` is interpreted the same as `bytes=1-`. I think both kind of errors should cause an HTTP error response.

> > Also, there is some duplication between matching the overall (incomplete) format via regex and then splitting into substrings.
> What do you mean here?

It's just that validation is essentially split between the regex and the string methods. So some of that is redundant. For example the regex would be unnecessary, if the parsing was bit more precise.

Reply:
> Additionally, how likely is this header to be malformed? How critical is it to get this precisely correct before the feature is merged at all? If it has to be perfect before it gets merged I’ll just close the PR and publish it as a shard.

I doesn't has to be perfect. But the basics should be covered and I think proper error handling is necessary for that.
As said below, I can take care of finishing it up and I expect this shouldn't take long to be merged.
The base implementation looks solid, just needs some adjustments to handle edge cases / errors properly.